pkg/k8shandler/service_monitor.go

```diff
@@ -0,0 +1,85 @@
+package k8shandler
+
+import (
+	"fmt"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/client/monitoring/v1"
+	v1alpha1 "github.com/openshift/elasticsearch-operator/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreateOrUpdateServiceMonitors ensures the existence of ServiceMonitors for Elasticsearch cluster
+func CreateOrUpdateServiceMonitors(dpl *v1alpha1.Elasticsearch) error {
+	serviceMonitorName := fmt.Sprintf("%s-%s", dpl.Name, "cluster")
+	owner := asOwner(dpl)
+
+	err := createOrUpdateServiceMonitor(serviceMonitorName, dpl.Namespace, dpl.Labels, owner)
+	if err != nil {
+		return fmt.Errorf("Failure creating service %v", err)
+	}
+
+	return nil
+}
+
+func createOrUpdateServiceMonitor(serviceMonitorName, namespace string, labels map[string]string, owner metav1.OwnerReference) error {
+	elasticsearchScMonitor := createServiceMonitor(serviceMonitorName, namespace, labels)
+	addOwnerRefToObject(elasticsearchScMonitor, owner)
+	err := sdk.Create(elasticsearchScMonitor)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("Failure constructing Elasticsearch ServiceMonitor: %v", err)
+	}
+
+	// TODO: handle update
+
+	return nil
+}
+
+func createServiceMonitor(serviceMonitorName, namespace string, labels map[string]string) *monitoringv1.ServiceMonitor {
```

Comment:
@lukas-vlcek @brancz Can you guys help create ServiceMonitor for elasticsearch?
I'm mostly guessing what the values should look like now.
Thank you.

Reply:
Shall we specify scrape interval here as well?

Reply:
scrape interval is configured globally for the stack, no need for it to be configurable here

Reply:
Let's go with the default, ok?